bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -2277,6 +2277,13 @@ private int computeOffsetAtLocation(ITextViewer textViewer, int x, int y) {
 	/** The width of the vertical ruler. */
 	protected static final int VERTICAL_RULER_WIDTH= 12;
 
+	/*
+	 * The colors of AbstractTextEditor can be configured by the "Text Editors"
+	 * preferences. This color configuration will be changed by theming. This means
+	 * no need for CSS basecolors here.
+	 */
+	private static final String DISABLE_CSS = "org.eclipse.e4.ui.css.disabled"; //$NON-NLS-1$
```

Comment:
here does this "magic" constant come from?

Reply:
From org.eclipse.e4.ui.css.swt: AbstractCSSSWTEngineImpl line 117